xbmc/cores/paplayer/AudioDecoder.cpp

```diff
@@ -58,7 +58,10 @@ void CAudioDecoder::Destroy()
   m_pcmBuffer.Destroy();
 
   if ( m_codec )
+  {
+    m_codec->DeInit();
```

Comment:
An alternative way can be to remove the:
```cpp
  // DeInit()
  // Should just cleanup anything as necessary.  No need to free buffers here if they
  // are allocated and destroyed in the destructor.
  virtual void DeInit()=0;
```
complete and to do this on destructors from `ADDON::CAudioDecoder` and like before on `VideoPlayerCodec`. What do you think about them?